src/chain/stateTransition/index.ts

```diff
@@ -8,35 +8,43 @@ import {
 } from "../../types";
 
 import {processBlock} from "./block";
-import {processEpoch, shouldProcessEpoch} from "./epoch";
-import {advanceSlot, cacheState} from "./slot";
+import {processEpoch} from "./epoch";
+import {processSlots} from "./slot";
+import {hashTreeRoot} from "@chainsafe/ssz";
+import  assert from "assert";
+
 
 export {
-  advanceSlot,
-  cacheState,
   processBlock,
   processEpoch,
 };
 
-export function executeStateTransition(state: BeaconState, block: BeaconBlock | null, verifyStateRoot = true): BeaconState {
-
-  if (block) {
-    while(state.slot < block.slot) {
-      cacheState(state);
-      if (shouldProcessEpoch(state)) {
-        processEpoch(state);
-      }
-      advanceSlot(state);
-    }
-    processBlock(state, block, verifyStateRoot);
-  } else {
-    cacheState(state);
-    if (shouldProcessEpoch(state)) {
-      processEpoch(state);
-    }
-    advanceSlot(state);
-  }
+//SPEC 0.7
+// def state_transition(state: BeaconState, block: BeaconBlock, validate_state_root: bool=False) -> BeaconState:
+//   # Process slots (including those with no blocks) since block
+// process_slots(state, block.slot)
+// # Process block
+// process_block(state, block)
+// # Validate state root (`validate_state_root == True` in production)
+// if validate_state_root:
+// assert block.state_root == hash_tree_root(state)
+// # Return post-state
+// return state
 
 
+export function stateTransition(
+  state: BeaconState, block: BeaconBlock,
+  validateStateRoot = false
+): BeaconState {
+  // Process slots (including those with no blocks) since block
+  processSlots(state, block.slot);
```

Comment:
Check tests, in current spec some of them are passing null blocks to advance slots

Reply:
should we not also play catchup with the while loop from `6.x` code?

Reply:
The changes that have been done according to 0.7.1 spec is-
* `validateStateRoot = false`  which was `true` before.
* added funtion `processSlots` - where there is the same while loop and equivalent code of `cacheState` and `advanceSlot ` 
* The only difference is that there is no if-else condition for checking null blocks. Instead there is an assert - `assert(state.slot <= slot)`
* Now what change should i made?

Reply:
That's fine, you should change spec test, where sending empty block was used to advance slot, to call processSlots method instead of stateTransition